### PR TITLE
feat: [PL-29416]: Support a checkbox in Notification config of usergr…

### PR DESCRIPTION
**Pull request by YounixIO**
…oups

#### Screenshots

Screenshot 2022-11-04 at 5 55 52 PM
Screenshot 2022-11-04 at 5 56 01 PM

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by YounixIO** [comment]
retrigger jest

**Comment by YounixIO** [comment]
retrigger cypress-cv

File: src/modules/20-rbac/components/NotificationList/NotificationList.tsx [hunk 1]
```diff
@@ -241,31 +252,61 @@ const ChannelRow: React.FC<ChannelRow> = ({
...
-                        disabled={edit}
-                      />
+                  <Container width="75%">
+                    <Container width="100%">
+                      <Layout.Horizontal spacing="small">
+                        <Container width="45%">
+                          <FormInput.Select
+                            name="type"
+                            placeholder={getString('common.selectAChannel')}
+                            items={edit ? options : notificationItems}
```
Comment (reetika04) [inline src/modules/20-rbac/components/NotificationList/NotificationList.tsx:262]: If it is disabled in edit mode, why are we passing options?

**Review by kapilkumawat86 — APPROVED** [review]

File: src/modules/20-rbac/components/NotificationList/NotificationList.tsx [hunk 2]
```diff
@@ -241,31 +252,61 @@ const ChannelRow: React.FC<ChannelRow> = ({
...
+                            {getNotificationByConfig(data).value}
+                          </Text>
+                        </Container>
                       </Layout.Horizontal>
+
+                      <div className={css.sendEmailToAllUsersContainer}>
+                        {formikProps.values.type === 'EMAIL' && (
+                          <FormInput.CheckBox
+                            name="sendEmailToAllUsers"
+                            label="Send email to all users part of the user group"
```
[the remark this answers is not in the document]
Comment (YounixIO) [reply src/modules/20-rbac/components/NotificationList/NotificationList.tsx:303]: Will move these to strings in a subsequent PR.

File: src/modules/20-rbac/components/NotificationList/NotificationList.tsx [hunk 3]
```diff
@@ -241,31 +252,61 @@ const ChannelRow: React.FC<ChannelRow> = ({
...
+                      </Layout.Horizontal>
                     </Container>
-                    <Container width="40%">{renderInputField(formikProps.values.type)}</Container>
-                  </>
+
+                    <div className={css.sendEmailToAllUsersContainer}>
+                      {formikProps.values.type === 'EMAIL' && (
+                        <FormInput.CheckBox
+                          name="sendEmailToAllUsers"
+                          label="Send email to all users part of the user group"
```
[the remark this answers is not in the document]
Comment (YounixIO) [reply src/modules/20-rbac/components/NotificationList/NotificationList.tsx:274]: Will move these to strings in a subsequent PR.

**Review by ishantmehtaharness — APPROVED** [review]
Looks Good

**Pull request merged**